File: RAI/AnimalExample/JavaImplementation/Dog.java
public class Dog extends Animal{
    private int size;
    private String pelaje;

    public Dog(String name, String specie, String breed, int size, String pelaje){
        super(name, specie, breed);
        this.size = size;
        this.pelaje = pelaje;
    }

    public void muerde(){
        System.out.println("El perro " + this.name + " muerde fuerte!!!");
    }

    public void makeSound(){
        System.out.println("El perro " + this.name + " hace guauuu!!...");
    }
    
    public void eat(){
        System.out.println("El perro " + this.name + " come...");
    }
    public void eat(String food){
        System.out.println("El perro " + this.name + " come " + food);
    }

    @Override
    public String toString(){
        return "Este perro se llama " + this.name + " de raza " + this.breed;
    }
}

File: RAI/AnimalExample/python_implementation/Dog.py
from Animal import Animal

class Dog(Animal):
    def __init__(self, name, specie, size, pelaje):
        super().__init__(name, specie)
        self.size = size
        self.pelaje = pelaje

    
    def make_sound(self):
        print("El perro " +self.name + " hace GUAUU...\n")

    def to_string(self):
        return "El perro " + self.name + " de pelaje " + self.pelaje

    def __str__(self):
        return "El perro " + self.name + " de pelaje " + self.pelaje